### DISCUSSION TITLE
`Error: Cannot access url.searchParams on a page with prerendering enabled` in +server.ts

### DISCUSSION THREAD

**Discussion by kesdigital** [sveltejs/kit#6849] [Q&A] [answered]
Am building a blog with mdsvex with an endpoint that can be called for a specific number of articles, with that number being specified as a query string, this works well in development with `prerender = true` set and the root `+layout.server.ts` but fails when I build for production.
```
Error: Cannot access url.searchParams on a page with prerendering enabled
```
Removing the `url.searchParams.get('limit')` line or setting `prerender = false` gets rid of the error, but this doesn't feel right, `+server.ts` can't be pre-rendered, why am I getting the error, is it a bug?.
```ts
export const GET: RequestHandler = async ({url, params}) => {
  try {
    const limitParam = url.searchParams.get('limit');
    const limit = limitParam ? parseInt(limitParam) : 6;
    const articles = await fetchArticles(limit);
    return json(articles);
  } catch (err) {
    console.log(err)
    throw error(500, 'An error occurred while fetching the posts')
  }
};
```
This endpoint is in it's own `routes/api/article` directory with no `+page.svelte` files.

## Package info
```
"@sveltejs/adapter-auto": "^1.0.0-next.75"
"@sveltejs/kit": "^1.0.0-next.483"
```

**Comment by teemingc** [comment, marked as answer]
From the docs:

> Note that you can still prerender pages that load data based on the page's parameters, such as a src/routes/blog/[slug]/+page.svelte route.
>
> Accessing [url.searchParams](https://kit.svelte.dev/docs/load#input-properties-url) during prerendering is forbidden. If you need to use it, ensure you are only doing so in the browser (for example in onMount).

https://kit.svelte.dev/docs/page-options#prerender-when-not-to-prerender

As a side note, you only receive the error during build because pre-rendering only occurs during build.
